splunk/Dockerfile

```diff
@@ -26,7 +26,7 @@ EXPOSE 8089
 
 USER root
 
-RUN mkdir -p /opt/splunk/var/lib/splunk
+RUN mkdir -p /opt/splunk/var/lib/splunk && pkill -f "splunkd"
```

Comment:
no way Splunk is running here if you didn't start it - I guess you miss -diagnosed some other problem.

Are you properly stopping Splunk in that pre-run script?

Reply:
The Process was running in the background. 
We already went over the fact that stopping Splunk manually during the first stage of the build, for some reason, fails the whole deployment at the end. reason unknown. that's why we've gone with the multistage in the first place. 
If I'd have to guess, the port probably stays occupied, but I can't remember if it was confirmed.

Reply:
The fact you end up having a process up means there is a bug in the build process that its not starting new containers for different build stages. 

We've gone with multistage so that any changes we don't want that are made in the build image do not make it to the final image.

The only thing that could mess things up for us here is if killing Splunk causes it to delete or corrupt the files we are copying to the final image. If that is the case, we can have a separate "clean" stage to kill the process. In any case this does not belong in a layer in the final image.

Reply:
This issue raise only when using `kube play kube`. my guess is both stages share the same network namespace, which is why the ports from the first stage stay occupied and the process linger. 
I did not find a different solution for it nor did I find a method to close Splunk using commands. 
I thought `RUN /bin/bash /opt/splunk/bin/splunk stop` will do the trick but it only works when I actually enter a Splunk container, otherwise it fails. 

I might resort to another layer just to kill the service, although I'd prefer not to...

Reply:
Solved! successfully closed the server from within pre-run